### PR TITLE
Feat/Added featured section

### LINKED ISSUE #82: As a CMS Editor I can add a single featured content to a Section
Sections should have a "Featured Content" field which allows for the selection of one page (article, survey, poll, quiz etc.) that is within that section. This is optional, a section doesn't have to have a featured content.

On the site, the featured content would then be displayed at the top of the section page before anything else (subsections, child articles).

### PR DESCRIPTION
**Pull request by sheralim012**
Fixes #82 

- Added featured content in section in wagtail admin
- Only 1 featured content can be added
- Featured content in the section isn't mandatory
- Added/updated section UI
- Used appropriate templates for models
- Added featured content, subsection, and articles in the section page UI
- Page chooser panel for featured content in the section will start from section
- Added test cases for page chooser panel queryset filtering

### CODE REVIEW

**Comment by sheralim012** [comment]
@dannyparsons section featured content page choose will start at section level, it will require some additional checks (added a TODO) but for now, this works as expected.

**Comment by sheralim012** [comment]
Hey @istride, could you please go through this PR.

File: home/wagtail_hooks.py [hunk 1]
```diff
@@ -38,3 +42,16 @@ def sort_footer_page_listing_by_path(parent_page, pages, request):
         pages = pages.order_by('path')
 
     return pages
+
+
+@hooks.register('construct_page_chooser_queryset')
+def limit_page_chooser(pages, request):
```
Comment (istride) [inline home/wagtail_hooks.py:48]: It is not clear enough to me what is happening in this function. There needs to be tests for this at the very least. Perhaps an explanatory comment. Ideally, we should only intervene in the case where we're selecting a page for a Section's featured content, otherwise, the queryset should not be touched.

**Review by istride — CHANGES_REQUESTED** [review]

File: home/models.py [hunk 2]
```diff
@@ -87,10 +87,17 @@ class Section(Page):
...
+            InlinePanel('featured_content', max_num=1, label=_("Featured Content")),
+        ], heading=_('Featured Content')),
     ]
 
     def get_context(self, request):
         context = super().get_context(request)
+        context['featured_content'] = [
+            featured_content.content for featured_content in self.featured_content.filter(content__live=True)
+        ]
+        context['sub_sections'] = self.get_children().type(Section)
```
[the remark this answers is not in the document]
Comment (sheralim012) [reply home/models.py]: done

File: home/wagtail_hooks.py [hunk 3]
```diff
@@ -38,3 +42,16 @@ def sort_footer_page_listing_by_path(parent_page, pages, request):
...
+    # TODO: add section check otherwise page chooser will break in other places like banner etc.
```
[the remark this answers is not in the document]
Comment (sheralim012) [reply home/wagtail_hooks.py:49]: this was implemented just for demonstration purposes as mentioned by Danny in a previous comment and TODO was added to discuss the approach. I think it would be better if you hope on a quick call for this as there a quite a few edge cases that I'm concerned about.

File: home/wagtail_hooks.py (hunk 1 above)
Comment (sheralim012) [reply home/wagtail_hooks.py:48]: I have added comments and tests and simplified some code kindly check @istride

**Comment by sheralim012** [comment]
Hey @istride, changes addressed.
